wcfsetup/install/files/lib/acp/page/ExceptionLogViewPage.class.php

```diff
@@ -141,15 +141,22 @@ public function readData() {
 		$this->calculateNumberOfPages();
 		
 		$i = 0;
-		
+		$seenHashes = [];
 		foreach ($this->exceptions as $key => $val) {
 			$i++;
+			
+			$parsed = ExceptionLogUtil::parseException($val);
+			if (isset($seenHashes[$parsed['stackHash']])) {
+				$parsed['collapsed'] = true;
+			}
+			$seenHashes[$parsed['stackHash']] = true;
```

Comment:
Did you intentionally put this before the page filtering in the following lines?

Reply:
Yes, because I want to deduplicate across pages. If you have a repeated error you don't need to see it on every single page of the log, just once is sufficient.

Reply:
I was just wondering because this approach would imply a sequential reading of all pages without skipping any page. But this isn't a real concern as the relevant exceptions can easily be de-collapsed.

Reply:
This was already happening previously. In fact the entire log file was always read into memory and then parsed into an array. If you have megabytes of logs you have bigger issues than a broken log viewer and should probably use FTP anyway.

Reply:
I meant literally reading/accessing the rendered pages by the user not reading the file via PHP.

Reply:
Ah, yes. I guess most admins will either read the log sequentially or use the exception ID search. And if not decollapsing is readily available.